src/maybe.jl

```diff
@@ -0,0 +1,21 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+export Maybe, MaybeVector
+
+"""
+Haskell-style [option type](https://en.wikipedia.org/wiki/Option_type) primarily used to allow optional inputs/outputs
+for functions.
+
+For example, in `src/Examples/docs_examples.jl`, we have `draw_cooketriplet(filename::Maybe{AbstractString} = nothing)`.
+If the function is called with a `filename` argument, then it is passed as a string, and `Vis.save(filename)` saves an
+image of the drawn example. If there is no argument, then the save call is dispatched to the empty `function
+Vis.save(::Nothing) end`. This allows the user to run the example without saving a file.
+"""
+const Maybe{T} = Union{T, Nothing}
```

Comment:
Why doesn't Julia have this by default..?

Reply:
In my opinion, it's essentially syntactic sugar, but for some reason it's controversial among the Julia community. There's an official recommendation for `Union{T, Nothing}` [here](https://docs.julialang.org/en/v1/manual/faq/#Nothingness-and-missing-values-1), a long discussion thread [here](https://discourse.julialang.org/t/aliases-for-union-t-nothing-and-union-t-missing/15402/37), and even an example of KristofferC trying to define `const ?{T} = Union{T, Nothing}` [here](https://github.com/JuliaWeb/GitHub.jl/pull/90/files#diff-f3f3eb3e7bb7977f66531d9d1f8855a7R2).

Reply:
After thinking about it more I'm leaning toward Union{T,Nothing}, without any type aliasing. Just used in place wherever necessary. It's slightly more verbose but extremely clear. 

For people not familiar with Haskell or similar languages Maybe{T} won't mean much - you'll have to look at the definition to figure it out. But Union{T,Nothing} is something any Julia programmer should understand.

In general we should avoid creating type aliases, macros, etc., that make Julia look like a different language. The code will be harder to read and understand for people who aren't part of the inner circle.